torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -1324,14 +1324,33 @@ def triton_config_reduction(size_hints, x, r, num_stages=1, num_warps=None) -> C
     while r < size_hints[1] and conditional_product(x, r) < target:
         r *= 2
 
-    cfg = {"XBLOCK": x, "RBLOCK": r}
     if num_warps is None:
         num_warps = conditional_product(x, r) // 128
     # On AMD GPU each warp has 64 lanes which is double the size on NV GPU,
     # therefore using half the number of warps here correspondingly.
     default_num_warps = 4 if torch.version.hip else 8
     min_num_warps = 1 if torch.version.hip else 2
     num_warps = next_power_of_2(min(max(num_warps, min_num_warps), default_num_warps))
+
+    # Check if maxGridSize is exceeded - if so then must scale XBLOCK further
+    max_grid_x = 4294967295 if torch.version.hip else 2147483647
+    warp_size = 64 if torch.version.hip else 32
+    num_blocks = int((size_hints[0] + x - 1) // x)
+    while (num_blocks * num_warps * warp_size) > max_grid_x:
+        if x >= TRITON_MAX_BLOCK["X"]:
+            if num_warps == 1:
+                break  # If no more scaling possible then break
```

Comment:
How about continue here and add an assert when `x` exceeds the max value so that user can tune up the max value? e.g,

https://github.com/pytorch/pytorch/blob/12d6446507df794e5f1f563250bbbd8bbd08044b/torch/_inductor/runtime/triton_heuristics.py#L1331

Reply:
Makes sense to me, this should simplify things allowing us to remove the scaling down of num_warps here.